src/Raven.Server/Smuggler/Documents/StreamSource.cs

```diff
@@ -708,12 +709,13 @@ private IEnumerable<DocumentItem> ReadDocuments(INewDocumentActions actions = nu
 
             var context = _context;
             var legacyImport = _buildVersionType == BuildVersionType.V3;
-            var modifier = new BlittableMetadataModifier(context)
+            var modifier = (BlittableMetadataModifier)context.RegisterModifierForCleanup(new BlittableMetadataModifier(context)
```

Comment:
Why do we have the `IBlittableMetadataModifier`, then? If there is just one impl?

Reply:
This is an excellent question

Reply:
I always wanted to kill `IBlittableMetadataModifier` but I was reluctant to do so because I didnt fully grasp why it was there in the first place. Probably a good idea to look into that because that causes slowdowns down the chain.